### ISSUE TITLE
No Spinner for NPM install

### ISSUE THREAD

**Issue by inspirebin** [babun/babun#403]
Usually when we type 'npm install' something, in terminal or powershell there is a spinner that indicates the progress of the file download. But in babun there is  no progress indicator 
![image](https://cloud.githubusercontent.com/assets/5962044/8892656/b120379a-3383-11e5-94e2-983cbcfa5a7e.png)

**Comment by tombujok** [comment]
hints anybody?

**Comment by Alex-D** [comment]
Same here. I don't see spinner in npm 2 or 3.

**Comment by Alex-D** [comment]
When I look the node `process.stdout` object, the `_type` is `pipe` when in others cmd I have `tty`.

So node don't see babun as graphical output but as a pipe :(

node don't get columns and row, so some node modules can't compute width of progressbar or something like that.

**Comment by Alex-D** [comment]
OK so, temp solution is to build `console.exe` from https://github.com/rprichard/winpty (win10 branch exists when I post this message). Then, add an alias to your zshrc like : 

`alias node="/c/dev/winpty/build/console.exe node"`

source the zshrc: it works! :)

**Comment by m0ngr31** [comment]
@Alex-D trying the same thing with `npm` gives an error: `Error 0x2 starting npm`. Any ideas?

**Comment by Alex-D** [comment]
Yep, I've rename node.exe into _node.exe and I've created node.exe as a bash script with something like:

`/c/dev/winpty/build/console.exe _node.exe`

npm, bower, gulp, grunt and other global command don't use the "node" command but the absolute path to node.exe so we need to change it :/

**Comment by rmobis** [comment]
Kinda of an old topic but if anyone does fall here as I did:

Downloading the latest [rprichard/winpty](https://github.com/rprichard/winpty/) release, unpacking it into cygwin's `bin` folder and setting these aliases worked for me:

```bash
alias node="winpty node"
alias npm="node C:/Program\ Files/nodejs/node_modules/npm/bin/npm-cli.js"
```

**Comment by nonoroazoro** [comment]
@rmobis Thanks a lot! Your solution works for me.

**Comment by yogeshlonkar** [comment]
@rmobis you are the BOSS
